backend/src/main/java/com/example/backend/repositories/ProductRepository.java

```diff
@@ -0,0 +1,50 @@
+package com.example.backend.repositories;
+
+import com.example.backend.models.Product;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class ProductRepository {
+
+    private static Map<Long, Product> products = new HashMap<>();
+    private static long currentId = 1L;
+
+    public Product save(Product product) {
+        product.setId(currentId++);
```

Comment:
this is not thread safe .. multiple products could be created with the same id loosing some of them in the map